marketdata/stream/client.go

```diff
@@ -360,6 +365,11 @@ func (c *client) maintainConnection(ctx context.Context, u url.URL, initialResul
 		if connectedAtLeastOnce {
 			close(c.terminatedChan)
 		}
+		// If a disconnect/connect callback is running then wait until it finishes or times out.
+		timeout := time.Second
+		if waitTimeout(&callbackWaitGroup, timeout) {
+			c.logger.Warnf("datav2stream: timed out after waiting one second for connect/disconnect callbacks to return")
```

Comment:
[minor] If the `timeout` is already in a variable, it should be used in the log as well.